core-bundle/src/Crawl/Escargot/Subscriber/BrokenLinkCheckerSubscriber.php

```diff
@@ -96,6 +96,13 @@ public function needsContent(CrawlUri $crawlUri, ResponseInterface $response, Ch
             return SubscriberInterface::DECISION_NEGATIVE;
         }
 
+        // Skip any redirected URLs that are now outside our base hosts (#4213)
+        $actualHost = parse_url($response->getInfo('url'), PHP_URL_HOST);
+
+        if ($crawlUri->getUri()->getHost() !== $actualHost && !$this->escargot->getBaseUris()->containsHost($actualHost)) {
+            return SubscriberInterface::DECISION_NEGATIVE;
+        }
+
         ++$this->stats['ok'];
```

Comment:
Shouldn’t `++$this->stats['ok'];` go above your new code block?

Reply:
@ausi no, it would then report, that x additional broken links have been checked. But in reality they were discarded.

Reply:
Hm, actually got confused there I think. Changed in 2a7072c48 .